ros2bag/ros2bag/verb/reindex.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2020 DCS Corporation, All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# DISTRIBUTION A. Approved for public release; distribution unlimited.
+# OPSEC #4584.
+#
+# Delivered to the U.S. Government with Unlimited Rights, as defined in DFARS
+# Part 252.227-7013 or 7014 (Feb 2014).
+#
+# This notice must appear in all copies of this file and its derivatives.
+
+from ros2bag.api import check_path_exists, print_error
+from ros2bag.reindexer import reindex_base
+from ros2bag.verb import VerbExtension
+
+
+class ReindexVerb(VerbExtension):
+    """Generate metadata from a bag."""
+
+    def add_arguments(self, parser, cli_name):  # noqa: D102
+        parser.add_argument(
+            'bag_file', type=check_path_exists, help='Bag file to reindex')
+        parser.add_argument(
+            '-s', '--storage-id', default='sqlite3',
+            help="storage identifier to be used, defaults to 'sqlite3'")
+        parser.add_argument(
```

Comment:
Both `compression-format` and `compression-mode` choices shouldn't be duplicated. Perhaps they should be factored out of `record.py` and this file into a common place - or you could at least pull the values out of `record.py`'s `add_argument` calls into a global level variable for that module, and `from ros2bag.verb.record import COMPRESSION_MODES` - something like that. The duplication will inevitable get desynchronized in the future